src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs

```diff
@@ -1,13 +1,16 @@
+using System;
 using System.Collections.Generic;
 using System.IO;
 using System.IO.Abstractions;
+using System.Linq;
 using Buildalyzer;
 using Stryker.Core.Initialisation.Buildalyzer;
+using Stryker.Core.InjectedHelpers;
 using Stryker.Core.ProjectComponents;
 
 namespace Stryker.Core.Initialisation
 {
-    public class ProjectInfo
+    public class ProjectInfo : IProjectAndTest
```

Comment:
I am afraid I have not looked into #1850. So definitely unaware of ongoing changes there.
Probably the biggest change I had to do was to remove direct references to the mutated project within most (if not all) Stryker classes. For example, VsTest runner expect to have the mutated project, as well as the relevant test projects as part of any test sessions. This is how I avoid to duplicate everything for each source projects.

Looking at TestProjectsInfo, I see there is no reference to SourceProjectInfo whatsoever. And this is ok, because in Solution mode, you end up with test projects being valid for multiple sourceprojects.

I am afraid an actual solution mode implies a lot of breaking changes to the #1850 PR.
I could replace `IProjectAndTest` with `ProjectInfo` if it helps, but I am not sure it will.

Reply:
Thinking a bit further, `IProjectAndTest` (which should `IProjectAndTests`) should be easy to implement as new class aggregating `SourceProjectInfo` and the appropriate `TestProjectInfo` (or `TestProjectsInfo`, not sure which is better).
In this PR, test projects are handled on a one by one basis to prevent duplicate operations as test project often covers multiple source projects.

Reply:
> Looking at TestProjectsInfo, I see there is no reference to SourceProjectInfo whatsoever. And this is ok, because in Solution mode, you end up with test projects being valid for multiple sourceprojects.

Yea I meant it the other way around.

> Thinking a bit further, IProjectAndTest (which should IProjectAndTests) should be easy to implement as new class aggregating SourceProjectInfo and the appropriate TestProjectInfo (or TestProjectsInfo, not sure which is better).

I think SourceProjectInfo could already be the aggregate (as it basically already was before the changes in #1850), not sure it's really necessary to create a new class for that purpose but we'll see.

> I am afraid an actual solution mode implies a lot of breaking changes to the https://github.com/stryker-mutator/stryker-net/pull/1850 PR.

Yeaaa this is gonna be fun 😅.. #1850 is ready to merge and I've barely started reviewing this PR so I think it makes the most sense to merge #1850 and then for me and @richardwerkman to fix all conflicts that causes in this PR because then it can act as the review at the same time. Makes sense to you?

Reply:
I am not sure what is the best way. I think it would be better if I try to do it myself. There is a significant chance that it turns out at extremely difficult to achieve

Reply:
As long as that's OK with you. Just trying to make sure we're not putting too much of a burden on you due to the order of merging.

Reply:
I am not sure which order is better, anyway. I still have a pretty good picture of the design changes I made.
So I guess merging this one post #1850 is the sensible approach
But whatever we look at it, this will be gruesome work for any one 😓

Reply:
status
code build
tests dont
does not work in normal mode
does not work in solution mode

conflicts are horrendous on test files and inputfileresolver

Reply:
I hope I can reach a status where tests compile and works my way out of here 😄

Reply:
Well, 49 failing tests. That's a start

Reply:
one fixed, only 48 to go. 🤣 .

Reply:
final countdown, three to go

Reply:
one remaining

Reply:
update: the failing test is the one captures a real issue, due to both branches having significantly changed the 'project discovery' design. 
I am not sure I will have enough free time this week to solve this (not that big a deal either)

Reply:
I'm at a conference so don't have much time either

Reply:
it now works for single project. solution mode still failing (as indicated by the failing test)

Reply:
fixed test and executions.
still refactoring to do though

Reply:
Great job on the quick merge and fixes!

Reply:
migration is complete. Note that there is no longer a `GetTestProjectInfo` method, only a `GetSourceProjectInfo`. This is due to how those are resolved depending on detection mode. The original _TestProject(s)_ mode, Stryker identifies the source project from the dependencies of the explicitly selected test projects. In _Solution_ mode, this is now the other way around: Stryker identifies source projects and their associated test projects. Hence, the initialization workflow can not assume which type of projects is needed 'first'.
And since mutated projects are the key entity here, it made sense to select them as the pivot values.